### PR TITLE
Routing: Remove subPath only at start of href

### LINKED ISSUE #35414: Routing: subpaths are matched regardless of location in href
**What happened**:
I used the subpath config option to add a base url to grafana. However that subpath is repeated in some urls which results in 404s.

https://user-images.githubusercontent.com/73201/121340478-f7977700-c91f-11eb-8211-cf0782551033.mp4

**What you expected to happen**:
To be able to navigate to the intended page.

**How to reproduce it (as minimally and precisely as possible)**:

1. Set the following in `custom.ini`
```
[server]
# The full public facing url
root_url = %(protocol)s://%(domain)s:%(http_port)s/grafana

# Serve Grafana from subpath specified in `root_url` setting. By default it is set to `false` for compatibility reasons.
serve_from_sub_path = true
```
2. Navigate to `/grafana/plugins` and click the `Install & Manage Plugins` button.

**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v8.1.0-pre (ec9b9d6ab1)
- Data source type & version: N/A
- OS Grafana is installed on: N/A
- User OS & Browser: Mac OSX Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
When stripping the subpath from a url it is matched against regardless of location in the href passed in (see issue for example). This PR changes that so it strips the subPath if it appears at the start of the href.

- [x] add tests

**Which issue(s) this PR fixes**:

Fixes #35414

**Special notes for your reviewer**:

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
Good catch @jackw

File: packages/grafana-data/src/utils/location.test.ts [hunk 1]
```diff
@@ -41,6 +41,14 @@ describe('locationUtil', () => {
       const urlWithoutMaster = locationUtil.stripBaseFromUrl('http://www.domain.com:9877/subUrl/grafana/');
       expect(urlWithoutMaster).toBe('/grafana/');
     });
+    test('relative url with subdirectory  subUrl', () => {
+      const urlWithoutMaster = locationUtil.stripBaseFromUrl('/grafana/subUrl');
+      expect(urlWithoutMaster).toBe('/grafana/subUrl');
+    });
```
Comment (torkelo) [inline packages/grafana-data/src/utils/location.test.ts:47]: this does not look correct, should not /grafana be stripped here ?

Comment (torkelo) [reply packages/grafana-data/src/utils/location.test.ts:47]: ah, subUrl is the real sub url here

**Review by torkelo — APPROVED** [review]

File: packages/grafana-data/src/utils/location.test.ts (hunk 1 above)
Comment (jackw) [reply packages/grafana-data/src/utils/location.test.ts]: Yeah this has been rather confusing for me. 😄 
I've tried to change these strings in the hope it makes the tests a little clearer (possibly not).

**Pull request merged**